### PR TITLE
Octo keygen library

### PR DESCRIPTION
**Pull request by ofiriluz**
Specify library name and version:  **octo-keygen-cpp/1.0.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

Keypair / certificates generation and encapsulation based on openssl

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/octo-keygen-cpp/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+            destination=self.source_folder, strip_root=True)
+
+    def requirements(self):
+        self.requires("octo-logger-cpp/1.1.0")
+        self.requires("octo-encryption-cpp/1.1.0")
+        self.requires("fmt/9.0.0")
+        self.requires("openssl/1.1.1q")
+
+    def build_requirements(self):
+        self.build_requires("cmake/3.24.0")
```
Comment (uilianries) [inline recipes/octo-keygen-cpp/all/conanfile.py:68]:
```suggestion
        self.tool_requires("cmake/3.24.0")
```

https://docs.conan.io/en/latest/migrating_to_2.0/recipes.html#requirements

**Review by uilianries — COMMENTED** [review]
If possible, also add a test_v1_package folder to validate Conan 1.x support. As real example, you can use: https://github.com/conan-io/conan-center-index/pull/12446/files#diff-350e084a0c88e5d0c468831be8cf219806281db0b59c2772798a6f8e48c46341

File: recipes/octo-keygen-cpp/all/conanfile.py (hunk 1 above)
Comment (ofiriluz) [reply recipes/octo-keygen-cpp/all/conanfile.py]: done and done

File: recipes/octo-keygen-cpp/all/test_v1_package/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+cmake_minimum_required(VERSION 3.8)
+project(test_package LANGUAGES CXX)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup(TARGETS)
+
+add_executable(${PROJECT_NAME} ../test_package/test_package.cpp)
+target_link_libraries(${PROJECT_NAME} PRIVATE CONAN_PKG::octo-keygen-cpp)
```
Comment (uilianries) [inline recipes/octo-keygen-cpp/all/test_v1_package/CMakeLists.txt:8]:
```suggestion
target_link_libraries(${PROJECT_NAME} PRIVATE octo-keygen-cpp::octo-keygen-cpp)
```
CONAN_PKG will almost deprecated

File: recipes/octo-keygen-cpp/all/test_v1_package/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,9 @@
+cmake_minimum_required(VERSION 3.8)
+project(test_package LANGUAGES CXX)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup(TARGETS)
+
```
Comment (uilianries) [inline recipes/octo-keygen-cpp/all/test_v1_package/CMakeLists.txt:6]:
```suggestion

find_package(octo-keygen-cpp CONFIG REQUIRED)

```
generated by cmake_find_package_multi

File: recipes/octo-keygen-cpp/all/test_v1_package/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
+from conans import ConanFile, CMake, tools
```
Comment (uilianries) [inline recipes/octo-keygen-cpp/all/test_v1_package/conanfile.py:1]:
```suggestion
from conans import ConanFile, CMake
from conan.tools.build import cross_building
```
Linter will not accept cross_building from conans

File: recipes/octo-keygen-cpp/all/test_v1_package/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    settings = "os", "arch", "compiler", "build_type"
+    generators = "cmake", "cmake_find_package_multi"
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def test(self):
+        if not tools.cross_building(self):
```
Comment (uilianries) [inline recipes/octo-keygen-cpp/all/test_v1_package/conanfile.py:15]:
```suggestion
        if not cross_building(self):
```

Comment (ofiriluz) [reply recipes/octo-keygen-cpp/all/test_v1_package/conanfile.py]: fixed :)

File: recipes/octo-keygen-cpp/all/test_v1_package/conanfile.py (hunk 4 above)
Comment (ofiriluz) [reply recipes/octo-keygen-cpp/all/test_v1_package/conanfile.py]: fixed :)

File: recipes/octo-keygen-cpp/all/test_v1_package/CMakeLists.txt (hunk 3 above)
Comment (ofiriluz) [reply recipes/octo-keygen-cpp/all/test_v1_package/CMakeLists.txt:6]: fixed :)

File: recipes/octo-keygen-cpp/all/test_v1_package/CMakeLists.txt (hunk 2 above)
Comment (ofiriluz) [reply recipes/octo-keygen-cpp/all/test_v1_package/CMakeLists.txt]: fixed :)

**Review by jgsogo — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGTM

**Pull request merged**